packages/rke2-ingress-nginx/rke2-ingress-nginx.patch

```diff
@@ -33,78 +33,93 @@ diff -x '*.tgz' -x '*.lock' -uNr packages/rke2-ingress-nginx/charts-original/tem
            imagePullPolicy: {{ .Values.controller.image.pullPolicy }}
          {{- if .Values.controller.lifecycle }}
+           lifecycle: {{ toYaml .Values.controller.lifecycle | nindent 12 }}
 diff -x '*.tgz' -x '*.lock' -uNr packages/rke2-ingress-nginx/charts-original/templates/controller-deployment.yaml packages/rke2-ingress-nginx/charts/templates/controller-deployment.yaml
 --- packages/rke2-ingress-nginx/charts-original/templates/controller-deployment.yaml
 +++ packages/rke2-ingress-nginx/charts/templates/controller-deployment.yaml
-@@ -65,7 +65,7 @@
+@@ -64,9 +64,7 @@
+     {{- end }}
        containers:
          - name: controller
-           {{- with .Values.controller.image }}
+-          {{- with .Values.controller.image }}
 -          image: "{{.repository}}:{{ .tag }}{{- if (.digest) -}} @{{.digest}} {{- end -}}"
-+          image: "{{ template "system_default_registry" . }}{{.repository}}:{{ .tag }}{{- if (.digest) -}} @{{.digest}} {{- end -}}"
-           {{- end }}
+-          {{- end }}
++          image: "{{ template "system_default_registry" . }}{{- with .Values.controller.image -}}{{.repository}}:{{ .tag }}{{- if (.digest) -}} @{{.digest}} {{- end -}}{{- end -}}"
            imagePullPolicy: {{ .Values.controller.image.pullPolicy }}
          {{- if .Values.controller.lifecycle }}
+           lifecycle: {{ toYaml .Values.controller.lifecycle | nindent 12 }}
 diff -x '*.tgz' -x '*.lock' -uNr packages/rke2-ingress-nginx/charts-original/templates/default-backend-deployment.yaml packages/rke2-ingress-nginx/charts/templates/default-backend-deployment.yaml
 --- packages/rke2-ingress-nginx/charts-original/templates/default-backend-deployment.yaml
 +++ packages/rke2-ingress-nginx/charts/templates/default-backend-deployment.yaml
-@@ -37,7 +37,7 @@
+@@ -36,9 +36,7 @@
+     {{- end }}
        containers:
          - name: {{ template "ingress-nginx.name" . }}-default-backend
-           {{- with .Values.defaultBackend.image }}
+-          {{- with .Values.defaultBackend.image }}
 -          image: "{{.repository}}:{{ .tag }}{{- if (.digest) -}} @{{.digest}} {{- end -}}"
-+          image: "{{ template "system_default_registry" . }}{{.repository}}:{{ .tag }}{{- if (.digest) -}} @{{.digest}} {{- end -}}"
-           {{- end }}
+-          {{- end }}
++          image: "{{ template "system_default_registry" . }}{{- with .Values.defaultBackend.image -}}{{.repository}}:{{ .tag }}{{- if (.digest) -}} @{{.digest}} {{- end -}}{{- end -}}"
            imagePullPolicy: {{ .Values.defaultBackend.image.pullPolicy }}
          {{- if .Values.defaultBackend.extraArgs }}
+           args:
 diff -x '*.tgz' -x '*.lock' -uNr packages/rke2-ingress-nginx/charts-original/values.yaml packages/rke2-ingress-nginx/charts/values.yaml
 --- packages/rke2-ingress-nginx/charts-original/values.yaml
 +++ packages/rke2-ingress-nginx/charts/values.yaml
-@@ -3,8 +3,8 @@
+@@ -1,11 +1,12 @@
++fullnameOverride: rke2-ingress-nginx
```

Comment:
We need to alter the name of the generated resources to be different from those created by the old version of the chart. Since RKE2 uses a release name of `rke2-ingress-nginx`, this doesn't give us that. Instead, lets append `v3` (for version 3.x of the chart) to the resource names:

```suggestion
+nameOverride: v3
```